libraries/botbuilder/tests/teamsActivityHandler.test.js

```diff
@@ -61,6 +61,226 @@ describe('TeamsActivityHandler', () => {
+            const bot = new InvokeActivity();
+
+            const adapter = new TestAdapter(async context => {
+                await bot.run(context);
+            });
+
+            const activity = createInvokeActivity('composeExtension/fetchTask');
+
+            adapter.send(activity)
+                .assertReply(activity => {
+                    assert.strictEqual(activity.value.status, 200, 'should be status code 200.')
+                    done();
+                })
+                .catch(err => done(err));
+        })
+
+        it('activity.name is "composeExtension/querySettingUrl". should return status code [200] when handleTeamsMessagingExtensionConfigurationQuerySettingUrl method is overridden.', done => {
+            const bot = new InvokeActivity();
+
+            const adapter = new TestAdapter(async context => {
+                await bot.run(context);
+            });
+
+            const activity = createInvokeActivity('composeExtension/querySettingUrl');
+
+            adapter.send(activity)
+                .assertReply(activity => {
+                    assert.strictEqual(activity.value.status, 200, 'should be status code 200.')
+                    done();
+                })
+                .catch(err => done(err));
+        })
+
+        it('activity.name is "composeExtension/setting". should return status code [200] when handleTeamsMessagingExtensionConfigurationSetting method is overridden.', done => {
+            const bot = new InvokeActivity();
+
+            const adapter = new TestAdapter(async context => {
+                await bot.run(context);
+            });
+
+            const activity = createInvokeActivity('composeExtension/setting');
+
+            adapter.send(activity)
+                .assertReply(activity => {
+                    assert.strictEqual(activity.value.status, 200, 'should be status code 200.')
+                    done();
+                })
+                .catch(err => done(err));
+        })
+
+        it('activity.name is "composeExtension/onCardButtonClicked". should return status code [200] when handleTeamsMessagingExtensionCardButtonClicked method is overridden.', done => {
+            const bot = new InvokeActivity();
+
+            const adapter = new TestAdapter(async context => {
+                await bot.run(context);
+            });
+
+            const activity = createInvokeActivity('composeExtension/onCardButtonClicked');
+
+            adapter.send(activity)
+                .assertReply(activity => {
+                    assert.strictEqual(activity.value.status, 200, 'should be status code 200.')
+                    done();
+                })
+                .catch(err => done(err));
+        })
+
+        it('should throw an error when onInvokeActivity param context is null.', done => {
+            class InvokeActivity2 extends InvokeActivity{

```

Comment:
This is another poor name and the inheritance from a subclass of TeamsActivityHandler opens the door for bugs in the test code if someone else modifies `InvokeActivity`.

Please rename the test class and extend TeamsActivityHandler, not the previous subclass.